libp2p/transports/tcptransport.nim

```diff
@@ -188,7 +204,20 @@ method accept*(self: TcpTransport): Future[Connection] {.async, gcsafe.} =
     raise newTransportClosedError()
 
   try:
-    let transp = await self.server.accept()
+    if self.acceptFuts.len <= 0:
+      self.acceptFuts = self.servers.mapIt(
+        it.accept()
+      )
+
+    if self.acceptFuts.len <= 0:
+      return
+
+    let finished = await one(self.acceptFuts)
+    self.acceptFuts.keepItIf(
+      it != finished
+    )
+
+    let transp = await finished
```

Comment:
This looks wrong to me, but I may be wrong

From what I understand, if we have two servers:
- We fill the acceptFuts with both accepts
- One accept eventually succeed, so we remove it from the `acceptFuts`
- On the next iteration `acceptFuts` only contains the second server, and since it's len is > 0 we don't re-add the first server
- So we only listen for the second server on the second iteration

I'll add a commit to fix this, but if your code was good feel free to revert mine